io-engine/src/bin/casperf.rs

```diff
@@ -130,12 +130,12 @@ impl Job {
             return;
         }
 
-        let offset = (job.rng.gen::<u64>() % job.io_size) * job.io_blocks;
+        let offset = job.rng.gen_range(0 .. job.io_blocks) * job.io_size;
```

Comment:
how do we ensure this doesn't span beyond device size?

Reply:
I believe that's ensured by the io_blocks calculation, which is the size of the device in io_size units.